Userland/Libraries/LibJS/Runtime/ObjectConstructor.cpp

```diff
@@ -76,6 +77,16 @@ JS_DEFINE_NATIVE_FUNCTION(ObjectConstructor::get_own_property_names)
     return Array::create_from(global_object, object->get_own_properties(PropertyKind::Key, false, GetOwnPropertyReturnType::StringOnly));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(ObjectConstructor::get_own_property_symbols)
+{
+    if (!vm.argument_count())
+        return {};
+    auto* object = vm.argument(0).to_object(global_object);
+    if (vm.exception())
+        return {};
+    return Array::create_from(global_object, object->get_own_properties(PropertyKind::Key, false, GetOwnPropertyReturnType::SymbolOnly));
+}
```

Comment:
This is exactly proving the point I was making earlier - the code which this is based on is broken in a subtle way, and as a result so is this - the argument count check is *not* part of the spec (see [here](https://tc39.es/ecma262/#sec-getownpropertykeys))! It was randomly added by someone *thinking* this could be a possible implementation, but turns out it's not :D

This is what I want to avoid going forward (also needs a section comment / link :wink:)

Reply:
The issue here is an empty value leak which will 100% blow something up. That's only a possible return value when we already have an exception and the value is discarded anyway.